src/components/learner-credit-management/BudgetAssignmentsTable.jsx

```diff
@@ -13,46 +13,138 @@ const BudgetAssignmentsTable = ({
   isLoading,
   tableData,
   fetchTableData,
-}) => (
-  <DataTable
-    isSortable
-    manualSortBy
-    isPaginated
-    manualPagination
-    isFilterable
-    manualFilters
-    isLoading={isLoading}
-    defaultColumnValues={{ Filter: TableTextFilter }}
-    FilterStatusComponent={FilterStatus}
-    columns={[
-      {
-        Header: 'Assignment details',
-        accessor: 'assignmentDetails',
-        Cell: AssignmentDetailsTableCell,
-        disableSortBy: true,
-      },
-      {
-        Header: 'Amount',
-        Cell: ({ row }) => `-${formatPrice(row.original.contentQuantity / 100, { maximumFractionDigits: 0 })}`,
-        disableFilters: true,
-      },
-    ]}
-    initialTableOptions={{
-      getRowId: row => row?.uuid?.toString(),
-    }}
-    initialState={{
-      pageSize: PAGE_SIZE,
-      pageIndex: DEFAULT_PAGE,
-      sortBy: [],
-      filters: [],
-    }}
-    fetchData={fetchTableData}
-    data={tableData.results}
-    itemCount={tableData.count}
-    pageCount={tableData.numPages}
-    EmptyTableComponent={CustomDataTableEmptyState}
-  />
-);
+}) => {
+  const [rowsSelected, setRowsSelected] = useState(false);
+
+  const CancelAction = ({ selectedFlatRows, ...rest }) => (
+    // Here is access to the selectedFlatRows, isEntireTableSelected, tableInstance
+    <Button variant="danger" onClick={() => console.log('Cancel', selectedFlatRows, rest)}>
+      {`Cancel (${selectedFlatRows.length})`}
+    </Button>
+  );
+
+  const RemindAction = ({ selectedFlatRows, ...rest }) => (
+    <Button onClick={() => console.log('Remind', selectedFlatRows)}>
+      {`Remind (${selectedFlatRows.length})`}
+    </Button>
+  );
+  
+  const ExtraAction = ({ text, selectedFlatRows, ...rest }) => (
+    <Button variant="outline-primary" onClick={() => console.log(`Extra Action ${text}`, selectedFlatRows, rest)}>
+      {`Refresh`}
+    </Button>
+  );
+
+  const rowSelected = (rowsSelected) => {
+    if (Object.keys(rowsSelected).length > 1) {
+      setRowsSelected(true);
+    } else {
+      setRowsSelected(false);
+    }
+  };
+
+  const renderBulkActions = (selectedFlatRows) => {
+    if (rowsSelected) {
+      return ([
+        <RemindAction />,
+        <CancelAction />,
+        <ExtraAction />,
```

Comment:
We don't need any kind of of ExtraAction, Remind and Cancel should be the only options available if rows are selected because refresh is no longer an option.

Reply:
Correct, the "Refresh" CTA is a table action, not a bulk action. It will only show when there are 0 selections, the default behavior of `DataTable`.